jerry-core/vm/vm.c

```diff
@@ -2922,6 +2922,21 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           ecma_fast_free_value (value);
           continue;
         }
+#if ENABLED (JERRY_ESNEXT)
+        case VM_OC_BRANCH_IF_NOT_UNDEFINED_NOR_NULL:
+        {
+          ecma_value_t value = *(--stack_top_p);
+
+          if (!ecma_is_value_null (value) && !ecma_is_value_undefined (value))
+          {
+            byte_code_p = byte_code_start_p + branch_offset;
+            ++stack_top_p;
+            continue;
+          }
+          ecma_fast_free_value (value);
```

Comment:
At this point `value` is neither undefined or null, so freeing this value is unnecessary.